services/slam/builtin/builtin.go

```diff
@@ -270,20 +271,20 @@ func (slamSvc *builtIn) GetPosition(ctx context.Context, name string) (spatialma
 
 // GetPointCloudMap creates a request, calls the slam algorithms GetPointCloudMap endpoint and returns a callback
 // function which will return the next chunk of the current pointcloud map.
-func (slamSvc *builtIn) GetPointCloudMap(ctx context.Context, name string) (func() ([]byte, error), error) {
+func (slamSvc *builtIn) GetPointCloudMap(ctx context.Context) (func() ([]byte, error), error) {
 	ctx, span := trace.StartSpan(ctx, "slam::builtIn::GetPointCloudMap")
 	defer span.End()
 
-	return grpchelper.GetPointCloudMapCallback(ctx, name, slamSvc.clientAlgo)
+	return grpchelper.GetPointCloudMapCallback(ctx, slamSvc.name, slamSvc.clientAlgo)
```

Comment:
Could consider sending the request instead of the name through to the grpchelper functions. If the number of arguments required for the request increases in the future it could lead to a more robust/simpler setup (less args to pass through), however since we current have only one input this is a non-issue.

That being said, it may still be advisable to centralize the req creation i in the `builtin.go` function instead of in the `grpchelper.go`

Reply:
I think how you have implemented it here makes sense.